### ISSUE TITLE
Crash in 1.21 when opening F3

### ISSUE THREAD

**Issue by codingcat2468** [tyrannus00/CleanF3#24]
If you open the F3 screen in Minecraft 1.21 with all options in the config enabled, the game crashes because of an `IndexOutOfBoundsException` in this mod.

![grafik](https://github.com/user-attachments/assets/fd6e46bb-d098-4422-979f-17881dec235e)
This issue most likely occures due to local difficulty being formatted differently when it can't be displayed (such as on servers).
In that case, the game replaces it with a "??" with causes an `IndexOutOfBoundsException` in this case, because the text is shorter than expected (in `text.subList()`):
![grafik](https://github.com/user-attachments/assets/b494edf1-4312-4657-82e1-c72d721e0d9d)

**Comment by tyrannus00** [comment]
I can't reproduce it unfortunately, even on servers where the local difficulty gets shown as `??`. The text.sublist... also references the lines, not the content per line. What server did the crash happen on, and does it also happen for you in singleplayer?

**Comment by codingcat2468** [comment]
Hi, I'm sorry for the confusion, seems like I wasn't paying much attention reading the code previously.
I took a second look at it and now actually found the issue. It seems like the issue is only happening in multiplayer because the noise router and detailed mob spawn data are only being shown on the F3 screen in singleplayer.
The line I was referencing before is normally supposed to delete this part of the F3 screen:

![grafik](https://github.com/user-attachments/assets/7efd70a0-91c1-4bb2-ba5b-2f2121775c8c)

Going into multiplayer, this part doesn't exist, which makes the code accidentally delete the wrong lines, in this case the sound info, active renderer info and the empty seperator line below:

![grafik](https://github.com/user-attachments/assets/0d263487-39d4-4e03-b29f-dbc5561d7279)

Now I'm also going to hide the active renderer info, which is deleted before this section of the code is being executed, resulting in it deleting this part of the F3 screen:

![grafik](https://github.com/user-attachments/assets/fbfaec27-0a4d-41ef-a2c4-4e4a9676330f)

The actual crash is going to happen if I now go ahead and also hide the help shortcut info, because the critical line of code will now try to delete this part of the screen:

![grafik](https://github.com/user-attachments/assets/f1a7160e-9ace-47cd-9319-649152f2a4e8)

The only remaining part here is the empty seperator line, and because the code trys to access the parts beyond that in the `subList()` call, an `IndexOutOfBoundsException` is thrown.

I'm also aware that this is really hard to find/reproduce since it requires a specific set of options activated at the same time (I only encountered this while playing a modpack that had all of the options of this mod enabled by default)!
